hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -3987,7 +3987,8 @@ public Object performTask() throws IOException {
               + " deleted and the block removal is delayed");
         }
         INodeFile iFile = ((INode)storedBlock.getBlockCollection()).asFile();
-        if (inodeIdentifier==null) {
+        if (inodeIdentifier == null || EntityManager.find(INode.Finder.ByINodeIdFTIS, inodeIdentifier.getInodeId())
```

Comment:
I think this check is already there in the parent function that calls it. It is part of the pull request that is bringing changes from EE to CE